pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -259,22 +261,45 @@ func (r *ReconcileKnativeServing) checkDeployments(instance *servingv1alpha1.Kna
 // Check for all dependencies
 func (r *ReconcileKnativeServing) checkDependencies(instance *servingv1alpha1.KnativeServing) error {
 	defer r.updateStatus(instance)
-
-	istio := schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1alpha3", Kind: "virtualservice"}
-	list := &unstructured.UnstructuredList{}
-	list.SetGroupVersionKind(istio)
-	if err := r.client.List(context.TODO(), nil, list); err != nil {
-		msg := fmt.Sprintf("Istio not detected; please install ServiceMesh")
-		instance.Status.MarkDependencyMissing(msg)
-		log.Error(err, msg)
-		return err
+	configuredMembersData := map[string]string{}
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	resource := &unstructured.Unstructured{}
+	resource.SetNamespace("istio-system")
+	resource.SetName("default")
+	resource.SetAPIVersion("maistra.io/v1")
+	resource.SetKind("ServiceMeshMemberRoll")
+	smmrData, err := r.config.Get(resource)
```

Comment:
I was thinking we'd use the unstructured stuff *only* if we weren't pulling in the maistra lib. Since we are, I think it may be clearer to just use the controller-runtime client, e.g.
```suggestion
	err := r.client.Get(context.TODO(), client.ObjectKey{Namespace: "istio-system", Name: "default"}, smmr)
```

Reply:
If i use controller-runtime client it only fetches the resources from `knative-serving` namespace
if we give different ns other than `knative-serving` it will give error saying not found.
That's the reason used `r.config.Get(resource)`

Reply:
Thank you for the suggestion on how to handle this part :)
Updated PR